docs/custom-extensions.md

```diff
@@ -0,0 +1,112 @@
+    restart. 
+
+## Adding custom extensions
+
+Custom extensions are downloaded by the Operator from the cloud storage. 
+User is in charge for properly packaging extension and uploading it to the storage.
+
+### Packaging custom extensions 
+
+Custom extension needs specific packaging to make the Operator able using it.
+The package must be a `.tar.gz` archive with all required files in a the correct
+directory structure.
+
+1. Control file must be in `SHAREDIR/extension` directory
+2. All required SQL script files must be in `SHAREDIR/extension` directory (there must be at least one SQL script)
+3. Any shared library must be in `LIBDIR` 
+
+!!! note 
+
+    In case of Percona Distribution for PostgreSQL images, `SHAREDIR` corresponds to `/usr/pgsql-${PG_MAJOR}/share` and `LIBDIR` to `/usr/pgsql-${PG_MAJOR}/lib`.
+
+For example, the directory for `pg_cron` extension should look as follows:
+
+``` {.bash data-prompt="$" }
+$ tree ~/pg_cron-1.6.1/
+/home/ege/pg_cron-1.6.1/
+└── usr
+    └── pgsql-15
+        ├── lib
+        │   └── pg_cron.so
+        └── share
+            └── extension
+                ├── pg_cron--1.0--1.1.sql
+                ├── pg_cron--1.0.sql
+                ├── pg_cron--1.1--1.2.sql
+                ├── pg_cron--1.2--1.3.sql
+                ├── pg_cron--1.3--1.4.sql
+                ├── pg_cron--1.4--1.4-1.sql
+                ├── pg_cron--1.4-1--1.5.sql
+                ├── pg_cron--1.5--1.6.sql
+                └── pg_cron.control
+```
+
+The archive must be created with `usr` at the root and the name must conform `${EXTENSION}-pg${PG_MAJOR}-${EXTENSION_VERSION}`:
+
+``` {.bash data-prompt="$" }
+$ cd pg_cron-1.6.1/
+$ tar -czf pg_cron-pg15-1.6.1.tar.gz usr/
+```
+
+!!! note
+
+    To understand which files are required for given extension could be not an easy task. One of the option to figure this out would be  building and installing the extension from source on a virtual machine with Percona Distribution for PostgreSQL and copy all the installed files to the archive.
+
+## Configuring custom extension loading
+
+When the extension is packaged, it should be uploaded to the cloud storage
+(for now, Amazon S3 is the only supported storage type). When the upload is done,
+the storage and extension details should be specified in the Custom Resource
+to make the Operator download and install it.
+
+Storage credentials are specified in `extensions.storage` subsection, and configuring it is similar to [setting Amazon S3 storage for backups](backups-storage.md):
+
+```yaml
+extensions:
+  ...
+  storage:
+    type: s3
+    bucket: pg-extensions
+    region: eu-central-1
+    secret:
+      name: cluster1-extensions-secret
+```
+
+When the storage is configured, and archive with extension is already present
+in the appropriate bucket, the extension itself can be specified to the Operator
+as follows:
+
+```yaml
+extensions:
+  ...
+  custom:
+  - name: pg_cron
+    version: 1.6.1
+```
+
```

Comment:
let's also include that the extension won't be enabled by default (it'll be just installed) and user is responsible of enabling it in desired databases using `CREATE EXTENSION pg_cron`.

some extensions also require to be added to `shared_preload_libraries` (such as pg_cron). Users can do it via cr.yaml:

```
  patroni:
    dynamicConfiguration:
      postgresql:
        parameters:
          shared_preload_libraries: pg_cron
```